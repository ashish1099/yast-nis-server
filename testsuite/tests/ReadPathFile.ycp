/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for the routines.ycp include
 *
 * Authors:
 *   Dan Vesely <dan@suse.cz>
 *
 * $Id$
 */

{
    include "testsuite.ycp";
    include "nis_server/io.ycp";

    map READ = $[
	"target" : $[
	    "size" : 42
	    ],
        "var" : $[
            "yp" : $[
                "makefile" : $[
                    "NOPUSH"        : "true",
                    "MERGE_PASSWD"  : "true",
                    "MERGE_GROUP"   : "true",
                    "MINUID"        : "100",
                    "MINGID"        : "100",
                    "maps"          : ["passwd", "group", "rpc", "services", "netid"]                            
                    ],
                "securenets" : [
                    $["netmask" : "255.0.0.0",
                      "network" : "127.0.0.0"
                        ]
                                
                    ],
                "ypservers" : []
                ]
            ],
        "rc" : $[
            "ypserver" : $[
                "YPPWD_CHFN"        : "no", 
                "YPPWD_CHSH"        : "no", 
                "YPPWD_SRCDIR"      : "/etc"
                ],
            "system" : $[
                "START_PORTMAP"       : "yes",
                "START_YPPASSWDD"     : "no",
                "START_YPSERV"        : "no",
                "START_YPXFRD"        : "no",
                "START_YPBIND"        : "yes",
                "YP_DOMAINNAME"       : "local"                
                ]
            ]
        ];


    DUMP ("YPMakefileVars");
    TEST (``(ReadPathFile (YPMakefileVars (), "/var/yp/Makefile")), [READ, $[], $[]], nil);
    DUMP ("YPServerVars");
    TEST (``(ReadPathFile (YPServerVars (), "/etc/rc.config.d/ypserv.rc.config")), [READ, $[], $[]], nil);
    DUMP ("YPSystemVars");
    TEST (``(ReadPathFile (YPSystemVars (), "/etc/rc.config")), [READ, $[], $[]], nil);
    DUMP ("YPSecurenets");
    TEST (``(ReadPathFile (YPSecurenets (), "/var/yp/securenets")), [READ, $[], $[]], nil);
    DUMP ("YPServers");
    TEST (``(ReadPathFile (YPServers (), "/var/yp/ypservers")), [READ, $[], $[]], nil);

}
