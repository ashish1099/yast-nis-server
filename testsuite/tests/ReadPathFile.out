Dump	YPMakefileVars
Read	.target.size "/var/yp/Makefile" 42
Read	.var.yp.makefile.MERGE_GROUP "true"
Read	.var.yp.makefile.MERGE_PASSWD "true"
Read	.var.yp.makefile.MINGID "100"
Read	.var.yp.makefile.MINUID "100"
Read	.var.yp.makefile.NOPUSH "true"
Read	.var.yp.makefile.maps ["passwd", "group", "rpc", "services", "netid"]
Return	$[".var.yp.makefile.MERGE_GROUP":"true", ".var.yp.makefile.MERGE_PASSWD":"true", ".var.yp.makefile.MINGID":"100", ".var.yp.makefile.MINUID":"100", ".var.yp.makefile.NOPUSH":"true", ".var.yp.makefile.maps":["passwd", "group", "rpc", "services", "netid"]]
Dump	YPServerVars
Read	.target.size "/etc/sysconfig/ypserv" 42
Read	.rc.ypserver.YPPWD_CHFN "no"
Read	.rc.ypserver.YPPWD_CHSH "no"
Read	.rc.ypserver.YPPWD_SRCDIR "/etc"
Return	$[".rc.ypserver.YPPWD_CHFN":"no", ".rc.ypserver.YPPWD_CHSH":"no", ".rc.ypserver.YPPWD_SRCDIR":"/etc"]
Dump	YPSystemVars
Read	.target.size "/etc/defaultdomain" 42
Read	.etc.defaultdomain "local"
Return	$[".etc.defaultdomain":"local"]
Dump	YPSecurenets
Read	.target.size "/var/yp/securenets" 42
Read	.var.yp.securenets [$["netmask":"255.0.0.0", "network":"127.0.0.0"]]
Return	$[".var.yp.securenets":[$["netmask":"255.0.0.0", "network":"127.0.0.0"]]]
Dump	YPServers
Read	.target.size "/var/yp/ypservers" 42
Read	.var.yp.ypservers []
Return	$[".var.yp.ypservers":[]]
